gdal/src/main/scala/geotrellis/contrib/vlm/gdal/effect/GDALRasterSource.scala

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.contrib.vlm.gdal.effect
+
+import geotrellis.contrib.vlm._
+import geotrellis.contrib.vlm.gdal.{GDALDataPath, GDALDataset, GDALWarpOptions}
+import geotrellis.contrib.vlm.effect._
+import geotrellis.contrib.vlm.effect.geotiff.UnsafeLift
+import geotrellis.contrib.vlm.avro._
+import geotrellis.proj4._
+import geotrellis.raster._
+import geotrellis.raster.io.geotiff.{AutoHigherResolution, OverviewStrategy}
+import geotrellis.raster.resample.{NearestNeighbor, ResampleMethod}
+import geotrellis.vector._
+import com.azavea.gdal.GDALWarp
+
+import cats._
+import cats.syntax.flatMap._
+import cats.syntax.apply._
+import cats.syntax.functor._
+
+case class GDALRasterSource[F[_]: Monad: UnsafeLift](
+  dataPath: GDALDataPath,
+  options: F[GDALWarpOptions] = GDALWarpOptions.EMPTY,
```

Comment:
@echeipesh it was a bit hard decision to make it like this, but actually we can't get around it :/